docs/reference_guide.md

```diff
@@ -870,6 +870,10 @@ Exactly one of `text` or `src_file` must be supplied (not both).
 
 The `cflags` specifies additional arguments to be passed to the compiler, for example `-DMACRO_NAME=value` or `-I/include/path`.  The arguments are passed as an array, with each element being an additional argument.  Note that strings are not split on whitespace, so each argument must be a different element of the array, e.g. `["-include", "header.h"]`.
 
+The special macro `BPF_LICENSE` specifies the license of the BPF program. Depending on which [BPF helpers](https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md#helpers) are used, a GPL-compatible license is required. **If the macro is not specified, BCC will automatically license the program as GPL.**
+
+To specify your own license, add the `cflags` argument `-DBPF_LICENSE=My License`. Note that it supports multiple words and quotes are not necessary. For instance, `cflags=['-DBPF_LICENSE=Dual BSD/GPL']` is a license that the kernel considers GPL-compatible (see the note below the reference of [BPF helpers](https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md#helpers)).
```

Comment:
The problem is that the license is forced in helpers.h, which I understand that is silently included by BCC. But because `helpers.h` is included before even processing the C code, it cannot be done from the source. Therefore it must be done from the command line.

I think what you ask could be done including a new `footer.h` header _after_ the code, in contrast to `helpers.h`, which is included _before_ the code. Maybe somebody else has a better solution.

Reply:
Another solution is just not specifying the license anywhere and let the user carry the burden, which is what I proposed in #1804.

Reply:
I prepared a new commit that sets the license at the end of the source. This way, the license macro can be set either in the code or as a CFLAGS define.

@4ast could you check the commit, before doing a pull request?

https://github.com/oriolarcas/bcc/blob/feature/footer-header/docs/reference_guide.md#1-bpf

Reply:
If we have a way to specify license in C file, maybe we do not need another way to specify through cflags? Specifying license in the C code is much cleaner/clearer as the license itself is associated with the file.

Reply:
Well, a macro can be defined in the command line or in the code. Would you prefer that I remove the CFLAGS comments from the documentation?

BTW, I want to point out the legal inconsistence between the Apache 2.0 license in the C headers and the actual license of the BPF program. I would consider relicensing the headers under LGPL or GPL with linking exception, otherwise it may be incompatible with the kernel's GPL 2.0 license. Should this happen in this new pull request?

Reply:
> Would you prefer that I remove the CFLAGS comments from the documentation?
Yes, if we can add license directly in the C file, I prefer not mentioning the way to do it through cflags.

Look at linux kernel, e.g., linux:tools/testing/selftests/bpf/sample_map_ret0.c.

```
/* SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause) */
#include <linux/bpf.h>
#include "bpf_helpers.h"
...
```
It has a license header but does not have license string.

or `sockmap_parse_prog.c`, it does not have license comments, but has
```
char _license[] SEC("license") = "GPL";
```
Yes, the bpf programs comment license probably should confirm to the license string. That is way everything related to license kept in the file is better. external cflags changing license may not change licenses in the comments...

In summary, I would like to do the same/similar way kernel does.

Reply:
Ok, now I see your point. I'll prepare the PR.